crates/bitwarden-crypto/src/enc_string/symmetric.rs

```diff
@@ -200,11 +237,38 @@ impl EncString {
         Ok(EncString::AesCbc256_HmacSha256_B64 { iv, mac, data })
     }
 
+    pub(crate) fn encrypt_xchacha20_poly1305(
+        data_dec: &[u8],
+        key: &XChaCha20Poly1305Key,
+    ) -> Result<EncString> {
+        let mut protected_header = coset::HeaderBuilder::new().build();
+        protected_header.alg = Some(coset::Algorithm::PrivateUse(cose::XCHACHA20_POLY1305));
+
+        let mut nonce = [0u8; 24];
+        let cose_encrypt0 = coset::CoseEncrypt0Builder::new()
+            .protected(protected_header)
+            .create_ciphertext(data_dec, &[], |data, aad| {
+                let ciphertext =
+                    crate::xchacha20::encrypt_xchacha20_poly1305(&(*key.enc_key).into(), data, aad);
+                nonce.copy_from_slice(ciphertext.nonce.as_slice());
```

Comment:
We should also be able to assing this without doing copy from slice by doing into(), which will also validate at compile time that both nonce fields are the same size.
```suggestion
                nonce = ciphertext.nonce.into();
```